src/Element/Element.php

```diff
@@ -165,9 +158,31 @@ public function findAll($selector, $locator)
         $xpath = $this->selectorsHandler->selectorToXpath($selector, $locator);
         $xpath = $this->xpathManipulator->prepend($xpath, $this->getXpath());
 
+        if ($firstOnly) {
+            $xpath = '(' . $xpath . ')[1]';
+        }
+
```

Comment:
I think, but not 100% certain, that this XPath altering code is better to be paced in classes, that builds these XPaths (e.g. the `\Behat\Mink\Selector\SelectorInterface::translateToXPath` methods).

Reply:
@aik099 those classes don't know whether you call `find` or `findAll`

Reply:
thus, they run before the prepending of the current element, so they would produce the wrong result.

Reply:
yes, private method to dedup the code